src/OSPSuite.SimModel/src/TableFormulaWithXArgument.cpp

```diff
@@ -0,0 +1,235 @@
+
+	//common object base part
+	ObjectBase::LoadFromXMLNode(pNode);
+
+	XMLNode pTableNode = pNode.GetChildNode(XMLConstants::TableObject);
+	_tableObjectId = (long)pTableNode.GetAttribute(XMLConstants::Id, INVALID_QUANTITY_ID);
+
+	XMLNode pXArgumentNode = pNode.GetChildNode(XMLConstants::XArgumentObject);
+	_XArgumentObjectId = (long)pXArgumentNode.GetAttribute(XMLConstants::Id, INVALID_QUANTITY_ID);
+}
+
+void TableFormulaWithXArgument::XMLFinalizeInstance (const XMLNode & pNode, Simulation * sim)
+{
+	const char * ERROR_SOURCE = "TableFormulaWithXArgument::XMLFinalizeInstance";
+
+	ObjectBase::XMLFinalizeInstance(pNode, sim);
+
+	_tableObject = sim->AllQuantities().GetObjectById(_tableObjectId);
+	if (_tableObject == NULL)
+		throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, "TableWithXArgument-Formula with id="+_idAsString+" references invalid table object with id "+XMLHelper::ToString(_tableObjectId));
+
+	long tableFormulaId = _tableObject->GetFormulaId();
+
+	if (tableFormulaId != INVALID_QUANTITY_ID)
+	{
+		Formula * formula = sim->Formulas().GetObjectById(tableFormulaId);
+
+		if (formula == NULL)
+			throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, "Object with id "+XMLHelper::ToString(_tableObjectId)+" references invalid formula");
+
+		_tableFormula = dynamic_cast<TableFormula *>(formula);
+		if (_tableFormula == NULL)
+			throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, "TableWithXArgument-Formula with id="+_idAsString+" references nontable object with id "+XMLHelper::ToString(_tableObjectId));
+	}
+	
+	_XArgumentObject = sim->AllQuantities().GetObjectById(_XArgumentObjectId);
+	if (_XArgumentObject == NULL)
+		throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, "TableWithXArgument-Formula with id="+_idAsString+" references invalid X argument object with id "+XMLHelper::ToString(_XArgumentObjectId));
+}
+
+bool TableFormulaWithXArgument::Simplify(bool forCurrentRunOnly)
+{
+	assert ((_tableObject != NULL) && (_XArgumentObject != NULL));
+
+	_tableObject->Simplify(forCurrentRunOnly);
+
+	//currently table object is defined by (x,y) data points and is NOT dependent from time or
+	//any parameters/variables
+	//Thus if the X argument object can be simplified (will be constant): 
+	//table formula with X argument can be simplified too
+	return  _XArgumentObject->Simplify(forCurrentRunOnly);
+}
+
+string TableFormulaWithXArgument::Equation()
+{
+	return "Table(X-Argument)"; //no explicit equation available
+}
+
+double TableFormulaWithXArgument::DE_Compute (const double * y, const double time, ScaleFactorUsageMode scaleFactorMode)
+{
+	assert ((_tableObject != NULL) && (_XArgumentObject != NULL));
+
+	double argument = _XArgumentObject->GetValue(y, time, scaleFactorMode);
+
+	//table formula always interprets the time passed to TableFormula::DE_Compute() as X argument
+	//thus here, just pass <argument> as a time parameter to _tableObject->GetValue
+	return _tableObject->GetValue(y, argument, scaleFactorMode);
```

Comment:
This is some abstraction leak here. My suggetion would be to have a special method for that in TableFormula or some other object where it makes sense

Reply:
You mean something like `TableFormula::GetValue(double argument)` ?